consul_kv.go

```diff
@@ -5,48 +5,40 @@ import (
 	"github.com/json-iterator/go"
 )
 
-var json = jsoniter.ConfigCompatibleWithStandardLibrary
-var kv *api.KV
-
 func NewConsulConfigStorage(consulAddress string) (*ConsulConfigStorage, error) {
 	//var err error
 	client, err := api.NewClient(&api.Config{Address: consulAddress})
 	if err != nil {
 		return nil, err
 	}
-
-	// Get a handle to the KV API
-	kv = client.KV()
-	return &ConsulConfigStorage{}, nil
+	json := jsoniter.ConfigCompatibleWithStandardLibrary
+	return &ConsulConfigStorage{kv: client.KV(), json: json}, nil
 }
 
 type ConsulConfigStorage struct {
 	config Config
+	kv     *api.KV
+	json   jsoniter.API
 }
 
 func (cs *ConsulConfigStorage) GetConfig(labName string) (Config, error) {
-	pair, _, err := kv.Get(labName, nil)
+	pair, _, err := cs.kv.Get(labName, nil)
 	if err != nil {
+		// get config failure from consul kv, cs.config default value is &Config{}
 		return cs.config, err
 	}
-	cs.config.Name = pair.Key
-	if err := json.Unmarshal(pair.Value, &cs.config.Treatments); err != nil {
+	if err := cs.json.Unmarshal(pair.Value, &cs.config.treatments); err != nil {
+		// unmarshal json failure, cs.config default value is &Config{}
 		return cs.config, err
 	}
+	cs.config.Name = pair.Key
 	return cs.config, nil
 }
 
-func (cs *ConsulConfigStorage) SetConfig(config Config) error {
+func (cs *ConsulConfigStorage) SetConfig(config Config) {
 	// PUT a new KV pair
-	value, err := json.Marshal(config.Treatments)
-	if err != nil {
-		return err
-	}
+	value, _ := cs.json.Marshal(config.treatments)
 	p := &api.KVPair{Key: config.Name, Value: value}
-	_, err = kv.Put(p, nil)
```

Comment:
Here, it seems a little bit weird to me, I think it should be kv.Put(config.Name, p)